Source/Core/DolphinQt2/Config/MappingWindow.cpp

```diff
@@ -0,0 +1,187 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <QComboBox>
+#include <QDialogButtonBox>
+#include <QGroupBox>
+#include <QHBoxLayout>
+#include <QPushButton>
+#include <QTabWidget>
+#include <QVBoxLayout>
+#include <iostream>
+
+#include "Core/Core.h"
+#include "Core/HW/GCPad.h"
+#include "DolphinQt2/Config/MappingWidget/GCPadWiiU.h"
+#include "InputCommon/ControllerEmu/ControllerEmu.h"
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "InputCommon/InputConfig.h"
+
+#include "DolphinQt2/Config/MappingWindow.h"
+
+MappingWindow::MappingWindow(const int portnr, QWidget* parent)
+    : QDialog(parent), controller(nullptr), port(portnr)
```

Comment:
This can be initialized within the class declaration as:

```cpp
class SomeName
{
...
  ControllerEmu::EmulatedController* controller = nullptr;
...
};
```

even if it's a pointer to a forward declared type.